### PR TITLE
Add support for mdc based correlation id for all log messages of single request flow

### PR DESCRIPTION
**Pull request by solita-alperttiti**

### CODE REVIEW

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/WebMvcConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import fi.thl.covid19.exposurenotification.error.CorrelationIdInterceptor;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class WebMvcConfiguration implements WebMvcConfigurer {
+
+    final
```
Comment (jyrkija) [inline exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/WebMvcConfiguration.java:11]: unnecessary row end

**Review by jyrkija — APPROVED** [review]
Already inspected in the old repository, but looked it through again. OK for merging.

File: exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/WebMvcConfiguration.java (hunk 1 above)
Comment (solita-alperttiti) [reply exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/WebMvcConfiguration.java:11]: fixed

File: exposure-notification/README.md [hunk 2]
```diff
@@ -48,6 +48,10 @@ When returning data from cache, those loggings are left out, so this should corr
 
 `MaintenanceService` logs numeric information about background tasks that clean up old data and pre-cache batches.
 
+In addition, for class level logging, Mapped Diagnostic Context (MDC) is used to add pseudorandom correlation identifier (correlationId)
+to every log message. CorrelationId binds all individual log messages of request flow together. For compatibility reasons,
+correlationId is called errorId in http-responses.
```
Comment (jyrkija) [inline exposure-notification/README.md:51]: The error id is mentioned above under ApiErrorHandler as well. That mention is now slightly deprecated - these should be combined. Same thing in the other README

Comment (solita-alperttiti) [reply exposure-notification/README.md:51]: ok, sanitized documentation a little bit.

**Review by jyrkija — APPROVED** [review]

**Pull request merged**